### ISSUE TITLE
Fire Stick 4k audio going in and out after a pause (AudioTrak RAW only)

### ISSUE THREAD

**Issue by ycodryn** [xbmc/xbmc#22230]
### Describe the bug
When watching a movie and pause it, after resume the video is playing but the audio is going in and out (stopping) every 2 sec. and sometimes the video goes without any sound and I have to restart kodi.

## Expected Behavior
Does not matter how many times I pause a movie, when resumed, the audio should be in sync with movie and not stopping every 1-2 sec.

### Debuglog

https://paste.kodi.tv/ocigazipof.kodi

### Your Environment
Used Operating system:

 - [X] Android

 - Operating system version/name:
 - Kodi version: v20

**Comment by tiagomiguel1994** [comment]
I'm having the same problem on nvidia Shield, if I pause the video the audio is stuttering, and if I pull forward or pull back it also stutters

**Comment by ycodryn** [comment]
using RC1 until now I did not have the problem again. I will further test and report back.

**Comment by ycodryn** [comment]
Until now with RC1 I had no problems. With RC2 the problem is back.

**Comment by tiagomiguel1994** [comment]
Here I always had problems with any version after Beta 2 and now with the latest version the problem persists.

**Comment by ycodryn** [comment]
Here is another log: https://paste.kodi.tv/vawabirada.kodi

With for example Euphoria series I don't have the problem, with The Head series the problem is back.

**Comment by tiagomiguel1994** [comment]
To me it seems to be sound in high bitrate files, like movies/series in DolbyTHD or HDR, in Kodi 19.4 everything works perfectly

**Comment by thexai** [comment]
This also happens if in Settings > System > Audio is selected **AudioTrak (IEC), Kodi IEC packer** in audio output device and passthrough output device?

**Comment by tiagomiguel1994** [comment]
Yes with IEC Kodi packer the problem disappeared completely.  
I had the RAW option selected. 
thanks

**Comment by ycodryn** [comment]
AudioTrak (IEC) is the only option and for the passtrough I'm using RAW because for some movies I don't have audio. I'm switching back to Kodi IEC to test it again.

**Comment by ycodryn** [comment]
Ok switched to Kodi IEC packer on passthrough and I had no sound at all on the same series that I had problems before. I turned off Dolby Digital capable receiver option and the sound is working now without any problems that I had before. So for now maybe this is the fix.
My only question is why Dolby digital option must be disabled? I know that firestick 4k is capable and also my soundbar.

**Comment by Nuklear92** [comment]
I wish this issue could be fixed in Android for the RAW AudioTrack option and also the one that seems to delay the audio to play whenever you pause or skip forward or backwards which has been since Kodi 19.0 if I'm correct. Because, sadly when using the AudioTrack IEC Kodi packer. There's random dropouts of audio when playing Dolby ATMOS tracks either in DD+ or TrueHD lossless.

**Comment by tiagomiguel1994** [comment]
Exactly, that's why I had the RAW option selected, because in IEC Kodi packer, I have audio dropouts

**Comment by Nuklear92** [comment]
> Exactly, that's why I had the RAW option selected, because in IEC Kodi packer, I have audio dropouts

Yeah, I wish this audio delay issues were fixed because that's the only issue with that. When it comes to playback of lossless audio tracks RAW plays them flawlessly without any random dropouts, etc.

**Comment by thexai** [comment]
In Shield right audio settings for TrueHD are:

Dolby processing = OFF
Match content resolution = ON
Audio video sync = 0 ms
Play comfort noise on HDMI = OFF

This is using "AudioTrack (IEC), Kodi IEC packer"

HDMI setup:
Shield ----> AVR/soundbar ----> TV

(NOT  Shield ----> TV ----> AVR/soundbar)

I wonder if they are using this Shield settings and HDMI setup.

I have Shield Pro 2019 (FW 9.0.2) and I don't have any issue with TrueHD.

**Comment by tiagomiguel1994** [comment]
Yes I have exactly the same settings, 
I will test to see if I have dropouts.  
However the bug in RAW could be fixed.

**Comment by fritsch** [comment]
Keep audio alive feature is broken since roughly a year now. I marked it as blocker of upcoming v20 - but I have no idea if it was planned to be fixed. With this broken, whenever you press "pause" and resume, audio sync will be gone - and in RAW case it needs several seconds to get back in sync.

You can btw. easily see that in the menus. Set keep audio alive to always, enable menu sound. Make sure you hear a click while moving. Now wait 15 seconds - next move won't make a click sound, but only the one after that. Same thing happens during playback.

There is another bug open for this issue.

**Comment by Nuklear92** [comment]
> In Shield right audio settings for TrueHD are:
> 
> 
> 
> Dolby processing = OFF
> 
> Match content resolution = ON
> 
> Audio video sync = 0 ms
> 
> Play comfort noise on HDMI = OFF
> 
> 
> 
> This is using "AudioTrack (IEC), Kodi IEC packer"
> 
> 
> 
> HDMI setup:
> 
> Shield ----> AVR/soundbar ----> TV
> 
> 
> 
> (NOT  Shield ----> TV ----> AVR/soundbar)
> 
> 
> 
> I wonder if they are using this Shield settings and HDMI setup.
> 
> 
> 
> I have Shield Pro 2019 (FW 9.1.1) and I don't have any issue with TrueHD. 
> 
> Don't have Fire Stick to test things.

These are the settings I have on my Shield since as long as I can remember. But the delayed audio comeback bug of RAW AudioTrack seems to be something different. And it's only recreational when you PAUSE, skip forward or backwards the content you're watching. The Audio will not come back even though the video is running, sometimes it comes back after like 10 seconds or more but mostly it just stay stuck in a guessing loop and doesn't come back unless you switch to a different audio track and back to the main you had selected if the content you're watching has multiple tracks from different codecs, etc.

**Comment by thexai** [comment]
> Keep audio alive feature is broken since roughly a year now.

Found various interesting things:

- Is broken when setting is "Always" but not with default "1 minute".
- Settings does not apply at change. Is need restart Kodi which is confusing.
- There is at least one overflow from trying to convert INT_MAX minutes to ms.
- Broken TrueHD and DD+ in AudioTrack RAW (this issue) has direct relation with audio keep alive but in a unexpected way...

PR is coming :)

**Comment by Nuklear92** [comment]
> > Keep audio alive feature is broken since roughly a year now.
> 
> 
> 
> Found various interesting things:
> 
> 
> 
> - Is broken when setting is "Always" but not with default "1 minute".
> 
> - Settings does not apply at change. Is need restart Kodi which is confusing.
> 
> - There is at least one overflow from trying to convert INT_MAX minutes to ms.
> 
> - Broken TrueHD and DD+ in AudioTrack RAW (this issue) has direct relation with audio keep alive but in a unexpected way...
> 
> 
> 
> PR is coming :)  

Wow! Awesome, thanks a bunch for this. I thought this would never be fixed as I had seen this issue ever since Kodi 19.0 I remember back in 18.x AudioTrack RAW was the default for Passthrough if I'm correct. Or at least the one that I always used, until I updated to Kodi 19.0 which then switched for using Kodi's own AudioTrack IEC Packer. Which doesn't have any delays for the audio returning normally when Pause or Skips but has issues where the Audio Drops out randomly when playback DD+/ATMOS or TrueHD with ATMOS. 

The only bad thing about this PR I saw, it's that it will be included into Kodi 21 Omega and not Nexus which is the next upcoming release.
😩 either way I truly appreciate this fix and we will definitely meet with it on the road to Omega later in the near future as they move forward to the Betas and RC releases. 🙂

Again, I highly appreciate the time for making this PR and maybe all the tests you needed to accomplish in order to made sure it's a working fix. 👍
